Allow workspace specific custom agents

JonasHelming opened the pull request:
fixed #15456

#### What it does

Allows workspace specific agents.
If workspace specific prompt locations are configured, the user can choose now whether to use the global prompt directory or a workspace directory for adding a custom agent. Workspace specific custom agents win. Within workspace locations, the first one loaded wins.

#### How to test

Add a project specific prompt directory (settings prompt templates)
Click on "Add custom agent" in the AI configuration view
Choose where you want to add the agent, it should show two locations now
Check that workspace specific custom agents "win"
Check updates when modifying, deleting and adding files.

#### Follow-ups

@sgraband Do you consider custom agents in your new view already?

#### Breaking changes

- [ ] This PR introduces breaking changes and requires careful review. If yes, the breaking changes section in the [changelog](https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md) has been updated.

#### Attribution

<!-- If the changelog entry for this change should contain an attribution at the end (e.g. Contributed on behalf of x) add it in this section -->

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

JonasHelming commented:
@eneufeld Ping!

eneufeld reviewed (commented):
Code looks fine. The feature is nice.
But I observe this behavior:
- I added customAgents.yml to my workspace .prompts folder and pressed `Add Custom Agent` button
- I don't get the quickpick but just my global file opens
- Only after restarting this was consumed
- Same in reverse, I deleted the customAgents.yml from the workspace and I still get the quickpick

JonasHelming commented:
> Code looks fine. The feature is nice. But I observe this behavior:
> 
> * I added customAgents.yml to my workspace .prompts folder and pressed `Add Custom Agent` button
> * I don't get the quickpick but just my global file opens
> * Only after restarting this was consumed
> * Same in reverse, I deleted the customAgents.yml from the workspace and I still get the quickpick

I cannot reproduce this.
Please note: You do not have to create customAgents.yml manually. the quick pick menu will show all possible locations. If a customAgents.yml is existing at that location it will open it. If there is no customAgents.yml, it will create one. This explains the "Same in reverse" issue you describe, this is actually expected. The first issue I cannot reproduce. I also tested this with manually creating a customAgents.yml in a new location, it worked for me.

eneufeld commented:
In the first case I don't get a quickpick. Did you modify any of the other settings, eg WorkSpace Template Directories? Do I have to manually add the .prompts folder there?

JonasHelming commented:
> In the first case I don't get a quickpick. Did you modify any of the other settings, eg WorkSpace Template Directories? Do I have to manually add the .prompts folder there?

It is not added by default, so yes, the very first time you have to add it.

eneufeld commented:
ok that helped.
PR works great, thanks for that.
One last question and maybe something for a followup:
What happens if the ids conflict? so if you have the same agent id in two customAgent.yml files which one is used in the prompt?
